io/GltfWriter.cpp

```diff
@@ -173,97 +204,165 @@ void GltfWriter::writeJsonChunk()
     j["asset"]["version"] = "2.0";
 
     j["buffers"].push_back(
-        {
-            { "byteLength", m_binSize }   // Total size of bin data.
-        }
+    {
+        { "byteLength", m_binSize }   // Total size of bin data.
+    }
     );
 
+    u_short elementSize =  sizeof(float) * 3; // X, Y, Z
+    if ( m_writeNormals ) {
+        elementSize += sizeof(float) * 3; // NormalX, NormalY, NormalZ
+    }
+    if ( m_colorVertices ) {
+        elementSize += sizeof(float) * 3; // R, G, B
+    }
+
     int bufferViewCount = 0;
+    u_short nextAccessorIndex = 0;
+    u_short normalAccessorIndex = 0;
+    u_short colorAccessorIndex = 0;
+    u_short positionAccessorIndex = 0;
+    u_short faceAccessorIndex = 0;
     for (const ViewData& vd : m_viewData)
     {
+        // Buffer views
+        // Vertex indices (faces)
+        const u_short faceBufferViewIndex = 0;
```

Comment:
I don't understand this change.  If there are more than one data view, all the views get combined.

Reply:
As far as I understand, the original version of this code before my changes created a separate BufferView for each ViewData, and a separate Accessor for each BufferView.

But then in in the actual primitive added to the gltf, it did this:
```
    NL::json mesh;
    mesh["primitives"].push_back(
        {
            { "attributes", {{"POSITION", 1}} },
            { "indices", 0 },
            { "material", 0 }
        }
    );
```
Each integer there is an index of an Accessor. Regardless of how many Accessors were actually created, it only ever references two: Accessor 0, which has the indices for the faces, and accessor 1, which describes the positions of the first BufferView. As far as I can see, all other accessors are ignored entirely, and therefore their geometry isn't actually rendered by an application when it loads the glTF. Indeed, a gltf validator will give warnings that accessors with indices > 1 aren't used. This seems incorrect to me, but I presumed it was like that for a reason, so I tried to make my code replicate this behaviour-- although I think I've done this wrong, and it will use the _last_ accessors created rather than the first ones.

I think a better way to do this would be to create a separate primitive for each ViewData, referencing the appropriate accessors for that ViewData. I'm happy to implement this, it just seemed beyond the scope of what I was trying to achieve with this PR.

Which would you prefer? I can update my code to keep working in a similar way, or I can change to code to make a primitive for each ViewData.

Reply:
I think you're probably correct.  My recollection is foggy.  I doubt that it was well tested and I found the terminology confusing.  Making separate primitives for each ViewData seems the way to go.  Thanks.